### ISSUE TITLE
Include projectRole and orgRole as part of `V1MemberUser`

### ISSUE THREAD

**Issue by lovincyrus** [rilldata/rill#5951]
Related to https://github.com/rilldata/rill/pull/5887

We need to display a tooltip for users who are viewers but have admin access within a user group. The tooltip should state, "This person can still edit because they are part of the group GROUP_NAME."

We have a couple of options: either specify project and organization roles in V1MemberUser[] or store an array of group names to indicate that the user has multiple access levels.

Option 1:
```diff
export interface V1MemberUser {
  userId?: string;
  userEmail?: string;
  userName?: string;
  userPhotoUrl?: string;
  roleName?: string; // deprecated
+ projectRole: string;
+ orgRole: { groupName: string, roleName: string }
  createdOn?: string;
  updatedOn?: string;
}
```

Option 2:
```diff
export interface V1MemberUser {
  userId?: string;
  userEmail?: string;
  userName?: string;
  userPhotoUrl?: string;
  roleName?: string;
+ multipleAccess?: ['Product', 'Marketing']
  createdOn?: string;
  updatedOn?: string;
}
```

As a workaround, we manually create a user group mapping from the user groups request to determine if the groups have admin access. Additionally, we'll use `createAdminServiceListUsergroupMemberUsers` to list users who are members of these groups, capturing those with access. See https://github.com/rilldata/rill/pull/5887/commits/0f62db884d27ddaabf7228ea82049d63e615b4e5

In summary, this feature addition will help us verify if a user, who is a viewer, also belongs to a group with admin access.

![CleanShot 2024-10-22 at 08 39 24@2x](https://github.com/user-attachments/assets/992f2a64-b9be-4923-af5d-17bed345f81f)

![CleanShot 2024-10-22 at 08 40 02@2x](https://github.com/user-attachments/assets/459bcaeb-8091-4e7f-a6d2-d049901668c7)

**Comment by begelundmuller** [comment]
Unfortunately this is not quite as simple as you outline in the note above. For example, someone may belong to an org-level group that has an org admin role, and our org-level admin roles currently propagate to projects.

In general, it's a bad idea to try and resolve permissions in the frontend because it's likely to either miss cases or get out of sync when the backend logic changes. It also requires potentially very large data that's not easily served into a frontend (1000s of groups is not that unlikely if an enterprise account sets up sync from an IDP, or if someone creates groups for partners using API access).

If necessary, we can provide the member's resolved permissions for the project (i.e. call the backend logic that resolves permissions), however each permissions resolution call is quite expensive, so I'm a little hesitant about doing that for many users at a time unless this is a high priority feature?
